pulsar-client-cpp/tests/MessageChunkSeekTest.cc

```diff
@@ -0,0 +1,124 @@
+#include "MessageIdImpl.h"
+
+DECLARE_LOG_OBJECT()
+
+using namespace pulsar;
+
+static const std::string lookupUrl = "pulsar://localhost:6650";
+
+// See the `maxMessageSize` config in test-conf/standalone-ssl.conf
+static constexpr size_t maxMessageSize = 1024000;
+
+static std::string toString(CompressionType compressionType) {
+    switch (compressionType) {
+        case CompressionType::CompressionNone:
+            return "None";
+        case CompressionType::CompressionLZ4:
+            return "LZ4";
+        case CompressionType::CompressionZLib:
+            return "ZLib";
+        case CompressionType::CompressionZSTD:
+            return "ZSTD";
+        case CompressionType::CompressionSNAPPY:
+            return "SNAPPY";
+        default:
+            return "Unknown (" + std::to_string(compressionType) + ")";
+    }
+}
+
+inline std::string createLargeMessage() {
+    std::string largeMessage(maxMessageSize * 3, 'a');
+    std::default_random_engine e(time(nullptr));
+    std::uniform_int_distribution<unsigned> u(0, 25);
+    for (size_t i = 0; i < largeMessage.size(); i++) {
+        largeMessage[i] = 'a' + u(e);
+    }
+    return largeMessage;
+}
+
+class MessageChunkingSeekTest : public ::testing::TestWithParam<CompressionType> {
+   public:
+    static std::string largeMessage;
+
+    void TearDown() override { client_.close(); }
+
+    void createProducer(const std::string& topic, Producer& producer) {
+        ProducerConfiguration conf;
+        conf.setBatchingEnabled(false);
+        conf.setChunkingEnabled(true);
+        conf.setCompressionType(GetParam());
+        LOG_INFO("Create producer to topic: " << topic
+                                              << ", compression: " << toString(conf.getCompressionType()));
+        ASSERT_EQ(ResultOk, client_.createProducer(topic, conf, producer));
+    }
+
+    void createConsumer(const std::string& topic, Consumer& consumer) {
+        ASSERT_EQ(ResultOk, client_.subscribe(topic, "my-sub", consumer));
+    }
+
+   private:
+    Client client_{lookupUrl};
+};
+
+std::string MessageChunkingSeekTest::largeMessage = createLargeMessage();
+
+TEST_P(MessageChunkingSeekTest, testSeek) {
+    const std::string topic =
+        "MessageChunkingSeekTest-testSeek-" + toString(GetParam()) + std::to_string(time(nullptr));
+    Consumer consumer;
+    createConsumer(topic, consumer);
+    Producer producer;
+    createProducer(topic, producer);
+
+    constexpr int numMessages = 20;
+
+    std::vector<MessageId> receiveMessageIds;
+    for (int i = 0; i < numMessages; i++) {
+        MessageId messageId;
+        ASSERT_EQ(ResultOk, producer.send(MessageBuilder().setContent(largeMessage).build(), messageId));
+        LOG_INFO("Send " << i << " to " << messageId);
+    }
+
+    Message msg;
+    for (int i = 0; i < numMessages; i++) {
+        consumer.receive(msg, 3000);
+        receiveMessageIds.push_back(msg.getMessageId());
+    }
+
+    consumer.seek(receiveMessageIds[1]);
```

Comment:
I see you just copied the `MessageChunkingTest.cc` except adding this line and below. There is too much repeated code. Could you just add the `testSeekChunkMessage` into `MessageChunkingTest`? And since this test is only responsible to verify whether the seek works, there is no need to test all the compression types. The test should be more simple.

See the similar test in Java client.

https://github.com/apache/pulsar/blob/7998c44b0b85e8ae1af5fec64d8f873032877a2f/pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChunkingTest.java#L453-L508